src/GraphQL.Tests/Complexity/ComplexityTestBase.cs

```diff
@@ -18,9 +20,16 @@ public class ComplexityTestBase
         public async Task<ExecutionResult> Execute(ComplexityConfiguration complexityConfig, string query) =>
             await StarWarsTestBase.Executer.ExecuteAsync(options =>
             {
-                options.Schema = new StarWarsTestBase().Schema;
+                options.Schema = CreateSchema();
                 options.Query = query;
                 options.ComplexityConfiguration = complexityConfig;
             });
+
+        //ISSUE: manually created test instance with ServiceProvider
+        private static StarWarsSchema CreateSchema()
+        {
+            var builder = new MicrosoftDI.GraphQLBuilder(new ServiceCollection(), b => new StarWarsTestBase().RegisterServices(b.Services));
+            return builder.ServiceCollection.BuildServiceProvider().GetRequiredService<StarWarsSchema>();
+        }
```

Comment:
I see you marked this and some other places with //ISSUE

Do you have a solution or for now is this just how it will be?

Reply:
Just a note about "not so good" implementation in my opinion. I don't want to rewrite those tests completely, not a goal for this PR. These tests create schema object that should be properly configured with registered services.